collect_app/src/androidTest/java/org/odk/collect/android/utilities/DateTimeUtilsTest.java

```diff
@@ -54,28 +55,28 @@ public void setUp() {
     }
 
     @Test
-    @Suppress
     public void getDateTimeLabelTest() {
-        long dateInMilliseconds = 687967200000L; // 20 Oct 1991 14:00
-
         Locale.setDefault(Locale.ENGLISH);
-        assertEquals("Oct 20, 1991", DateTimeUtils.getDateTimeLabel(new Date(dateInMilliseconds), gregorianDatePickerDetails, false, context));
-        assertEquals("Oct 20, 1991, 14:00", DateTimeUtils.getDateTimeLabel(new Date(dateInMilliseconds), gregorianDatePickerDetails, true, context));
+        TimeZone.setDefault(TimeZone.getTimeZone("GMT"));
 
-        Locale.setDefault(Locale.ENGLISH);
-        assertEquals("9 Tikimt 1984 (Oct 20, 1991)", DateTimeUtils.getDateTimeLabel(new Date(dateInMilliseconds), ethiopianDatePickerDetails, false, context));
-        assertEquals("9 Tikimt 1984, 14:00 (Oct 20, 1991, 14:00)", DateTimeUtils.getDateTimeLabel(new Date(dateInMilliseconds), ethiopianDatePickerDetails, true, context));
+        // 20 Oct 1991 14:00 GMT
+        Calendar calendar = Calendar.getInstance();
```

Comment:
I think we could use `Calendar.getInstance(TimeZone.getTimeZone("GMT"))` instead (using the timezone'd version of [getInstance()](https://developer.android.com/reference/java/util/Calendar.html#getInstance(java.util.TimeZone))). That lets us avoid changing the default time zone (with `TimeZone.setDefault`) for the process which I'm worried might pollute other tests.

Reply:
I tried that as well. But it didn't work as expected. The reason being that `DateFormatter` which is used in `DateTimeUtils` uses the default timezone (device's timezone instead of GMT) if nothing is provided to it. 
So `Calendar.getInstance(TimeZone.getTimeZone("GMT"))` solves the expecting date format but not the actual code which formats the date and time

Reply:
Ah right yeah that's annoying! Maybe I'm going to far here but I'm wondering if we can set the time zone onto the formatter. This way the method would always return a formatted date in the timezone of the `Date` passed to it.

I messed around with this and found you can do this:

```java
Calendar instance = Calendar.getInstance();
instance.setTime(date);
dateFormatter.setTimeZone(instance.getTimeZone());
```

Seems like we're running into complications with `Date` mostly being a deprecated interface here. What do you think?

Reply:
I agree, adding another parameter to set timezone to the DateTimeUtils is the solution.

Reply:
@seadowg @grzesiek2010 Should I add timezone to the `DateTimeUtils` for the test case or is it just fine like this?

Reply:
I'm not sure you'll need to add a parameter for `TimeZone` as you should be able to grab the zone from the provided `Date` using the (admittedly awkward) example above.  You could do this in `getGregorianDateTimeLabel` before the `format` call for instance.

If that feels too awkward due to all the interactions with multiple `DateFormatter` instances in `DateTimeUtils` it could be more pragmatic to have the test store the original default `TimeZone` and then reset it back in an `@After` to prevent this test polluting any others.

Reply:
@seadowg Done

Reply:
To verify that it works, I set the locale and timezone of my device to non-defaults (hi, Asia/Kolkata) and then printed the outputs of `Locale.getDefault()` and `Timezone.getDefault()` before and after the test and it matched as you suggested